pkg/endpoint/endpoint_status_test.go

```diff
@@ -0,0 +1,251 @@
+	e := newEndpoint(c, endpointGeneratorSpec{})
+	cepA := e.GetCiliumEndpointStatus()
+
+	go func() {
+		for i := 0; i < 1000; i++ {
+			e.Status.addStatusLog(&statusLogMsg{
+				Status: Status{Code: OK, Msg: "Success", Type: BPF},
+			})
+			time.Sleep(time.Millisecond)
+		}
+	}()
+
+	// Generate EndpointStatus in quick interval while state transitions
+	// are succeeding in the background
+	timeout := time.After(1 * time.Second)
+	tick := time.Tick(10 * time.Millisecond)
+	for {
+		select {
+		case <-timeout:
+			return
+		case <-tick:
+			cepB := e.GetCiliumEndpointStatus()
+			c.Assert(cepA, checker.DeepEquals, cepB)
+		}
+	}
+}
+
+func (s *EndpointSuite) TestGetCiliumEndpointStatusDeepEqual(c *check.C) {
+	a := newEndpoint(c, endpointGeneratorSpec{
+		fakeControllerManager:    true,
+		failingControllers:       10,
+		logErrors:                maxLogs,
+		allowedIngressIdentities: 100,
+		allowedEgressIdentities:  100,
+		numPortsPerIdentity:      10,
+	})
+
+	b := newEndpoint(c, endpointGeneratorSpec{
+		fakeControllerManager:    true,
+		failingControllers:       10,
+		logErrors:                maxLogs,
+		allowedIngressIdentities: 100,
+		allowedEgressIdentities:  100,
+		numPortsPerIdentity:      10,
+	})
+
+	cepA := a.GetCiliumEndpointStatus()
+	cepB := b.GetCiliumEndpointStatus()
+
+	c.Assert(cepA, checker.DeepEquals, cepB)
+}
+
+func (s *EndpointSuite) TestGetCiliumEndpointStatusCorrectnes(c *check.C) {
+	e := newEndpoint(c, endpointGeneratorSpec{
+		fakeControllerManager:    true,
+		failingControllers:       10,
+		logErrors:                maxLogs,
+		allowedIngressIdentities: 100,
+		allowedEgressIdentities:  100,
+		numPortsPerIdentity:      10,
+	})
+
+	cep := e.GetCiliumEndpointStatus()
+
+	c.Assert(len(cep.Status.Log), check.Equals, cilium_v2.EndpointStatusLogEntries)
+}
+
+func (s *EndpointSuite) BenchmarkGetCiliumEndpointStatusDeepEqual(c *check.C) {
+	a := newEndpoint(c, endpointGeneratorSpec{
+		fakeControllerManager:    true,
+		failingControllers:       10,
+		logErrors:                maxLogs,
+		allowedIngressIdentities: 100,
+		allowedEgressIdentities:  100,
+		numPortsPerIdentity:      10,
+	})
+
+	b := newEndpoint(c, endpointGeneratorSpec{
+		fakeControllerManager:    true,
+		failingControllers:       10,
+		logErrors:                maxLogs,
+		allowedIngressIdentities: 100,
+		allowedEgressIdentities:  100,
+		numPortsPerIdentity:      10,
+	})
+
+	c.ResetTimer()
+	for i := 0; i < c.N; i++ {
+		if !reflect.DeepEqual(a, b) {
+			c.Errorf("DeepEqual failed")
+		}
+	}
+	c.StopTimer()
+}
+
+func (s *EndpointSuite) BenchmarkGetCiliumEndpointStatus(c *check.C) {
+	e := newEndpoint(c, endpointGeneratorSpec{
+		failingControllers:       10,
+		logErrors:                maxLogs,
+		allowedIngressIdentities: 100,
+		allowedEgressIdentities:  100,
+		numPortsPerIdentity:      10,
+	})
+
+	c.ResetTimer()
+	for i := 0; i < c.N; i++ {
+		status := e.GetCiliumEndpointStatus()
+		c.Assert(status, check.Not(check.IsNil))
```

Comment:
Won't having assertions within a benchmark distort the benchmark a bit?

Reply:
A tiny bit but it's essentially free compared to all of the other operations. The purpose of the benchmark is to give a rough estimate on how many endpoint status operations we can perform per second.